src/pkg/ocidriver/ocirepository.go

```diff
@@ -0,0 +1,257 @@
+	"github.com/docker/distribution/manifest/schema2"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/distribution/registry/client"
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+	"io"
+	"sort"
+)
+
+// OciRepository ...
+type OciRepository struct {
+	Registry *OciRegistry
+
+	Name string
+}
+
+// List ...
+func (rp *OciRepository) List( /*filter*/ ) ([]*OciImage, error) {
+	repoNameRef, err := reference.WithName(rp.Name)
+	if err != nil {
+		return nil, errors.New("bad repository " + rp.Name)
+	}
+	repo, err := client.NewRepository(repoNameRef, rp.Registry.baseURL, rp.Registry.transport)
+	tags, err := repo.Tags(nil).All(nil)
+
+	sort.Strings(tags)
+	var ts []*OciImage
+	for _, tag := range tags {
+		ts = append(ts, &OciImage{
+			Registry:   rp.Registry,
+			repository: rp,
+			Name:       tag,
+		})
+	}
+	return ts, nil
+}
+
+// GetBlob ...
+func (rp *OciRepository) GetBlob(d digest.Digest) (size int64, data io.ReadCloser, err error) {
+	repoNameRef, err := reference.WithName(rp.Name)
+	if err != nil {
+		return 0, nil, err
+	}
+	repo, err := client.NewRepository(repoNameRef, rp.Registry.baseURL, rp.Registry.transport)
+	if err != nil {
+		return 0, nil, err
+	}
+	b, err := repo.Blobs(nil).Open(nil, d)
+	if err != nil {
+		return 0, nil, err
+	}
+	size, err = b.Seek(0, io.SeekEnd)
+	if err != nil {
+		return 0, nil, err
+	}
+	_, err = b.Seek(0, io.SeekStart)
+	return size, b, err
+}
+
+// GetImageByTag ...
+func (rp *OciRepository) GetImageByTag(tag string) (isList bool, manifest *OciManifest, manifestList *OciManifestIndex, manifestv1 *OciManifestv1, err error) {
+	repoNameRef, err := reference.WithName(rp.Name)
+	if err != nil {
+		return false, nil, nil, nil, err
+	}
+	ref, err := reference.WithTag(repoNameRef, tag)
+	if err != nil {
+		return false, nil, nil, nil, err
+	}
+	return rp.getImage(ref)
+}
+
+// GetImageByDigest ...
+func (rp *OciRepository) GetImageByDigest(d digest.Digest) (isList bool, manifest *OciManifest, manifestList *OciManifestIndex, manifestv1 *OciManifestv1, err error) {
+	repoNameRef, err := reference.WithName(rp.Name)
+	if err != nil {
+		return false, nil, nil, nil, err
+	}
+	ref, err := reference.WithDigest(repoNameRef, d)
+	if err != nil {
+		return false, nil, nil, nil, err
+	}
+	return rp.getImage(ref)
+}
+
+func (rp *OciRepository) getImage(ref reference.Named) (isList bool, manifest *OciManifest, manifestList *OciManifestIndex, manifestv1 *OciManifestv1, err error) {
+	repoNameRef, err := reference.WithName(rp.Name)
+	if err != nil {
+		return false, nil, nil, nil, err
+	}
+	repo, err := client.NewRepository(repoNameRef, rp.Registry.baseURL, rp.Registry.transport)
+	if err != nil {
+		return false, nil, nil, nil, err
+	}
+	ms, err := repo.Manifests(nil)
+	if err != nil {
+		return false, nil, nil, nil, err
+	}
+	var m1 distribution.Manifest
+	if digested, isDigested := ref.(reference.Canonical); isDigested {
+		m1, err = ms.Get(nil, digested.Digest())
```

Comment:
Set the accepting manifest media type here, or the registry will return the manifest as schema1.